JamieMason/syncpack
# Allow multiple filters to be specified for compatible syncpack sub commands

## Description (What)

Allows multiple filters to be specified for compatible syncpack sub commands. So to fix mismatches for multiple dependencies we can do something like:

```sh
syncpack fix-mismatches -f "@apollo" -f "webpack"
```

I also specified an `.nvmrc` with the latest node version since we use something a little more outdated and had to pick something. I would be happy to change this to the node version you are using instead!

## Justification (Why)

At QuickBase we had a use case where we had a monorepo with different dependency versions and combinations and needed to globally update all of these packages. It was easiest to get a list of all the packages in the repository using `list-mismatches` then be able to run `fix-mismatches` in one go rather than scripting a for-loop for calling `fix-mismatches` for each dependency that needed to be fixed.

For example:
```
.
├── packages
│   ├── A
│   ├── B
│   ├── C
```

Package A has dependencies `webpack`, `url-loader` and, `css-loader`
Package B has dependencies `webpack` and, `css-loader`
Package C has dependencies `@apollo`, `webpack` and, `css-loader`

To fix all of the mismatches at once I could call:
```
syncpack fix-mismatches -f "webpack" -f "@apollo" -f "url-loader" -f "css-loader"
```

## How Can This Be Tested?

- ensure you have node version >=12.0.0 for eslint compatibility
- run 
  ```sh
  yarn install
  yarn test
  yarn build
  ```
- go to another mono repository with multiple packages, ensure that you multiple filters are allowed for all compatible syncpack subcommands
